kernel/fs/ext2/mod.rs

```diff
@@ -0,0 +1,140 @@
+#![cfg_attr(not(test), no_std)]
+#![allow(unused)]
+use crate::result::Result;
+use core::ops::{Deref, DerefMut};
+
+pub trait Disk {
+    fn read_block(&mut self, lba: usize, buf: &mut [u8]) -> Result<usize>;
+    fn write_block(&mut self, lba: usize, buf: &[u8]) -> Result<usize>;
+}
+
+pub struct Ext2Fs<D: Disk> {
+    disk: D,
+}
+
+impl<D: Disk> Ext2Fs<D> {
+    pub fn new(disk: D) -> Ext2Fs<D> {
+        Ext2Fs { disk }
+    }
+}
+
+impl<D: Disk> Deref for Ext2Fs<D> {
+    type Target = D;
+
+    fn deref(&self) -> &Self::Target {
+        &self.disk
+    }
+}
+
+impl<D: Disk> DerefMut for Ext2Fs<D> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.disk
+    }
+}
```

Comment:
Why do you want to implement `Deref<Target=D>`? I think it's not common in Rust (instead of adding an accessor method like `Ext2Fs::disk(&self) -> &Disk`.

Reply:
In my understanding, whether it is the ext2 file system or other file systems, they all have Disk attributes. So I use `Deref` and `DerefMut` to make the ext2 object behave like a Disk to the outside. 
This seems to be a wrong understanding